src/loki/matrix_action.cc

```diff
@@ -105,6 +103,15 @@ void loki_worker_t::matrix(Api& request) {
   auto& options = *request.mutable_options();
   const auto& costing_name = Costing_Enum_Name(options.costing_type());
 
+  // Validate distance if disable hiererchy pruning is true
+  auto costing_options = options.mutable_costings()->find(options.costing_type());
+  if (costing_options != options.costings().end() &&
+      costing_options->second.options().disable_hierarchy_pruning() &&
+      !check_hierarchy_distance(options.locations())) {
+    add_warning(request, 205);
+    costing_options->second.mutable_options()->set_disable_hierarchy_pruning(false);
+  }
```

Comment:
at this point in the processing, you already went through the `ParseBaseCostingOptions` so the costing options are set, no need to check that again here. also you can put this whole function into the body of the function you added to loki::worker, right now it's not DRY and you repeat the same thing in all actions that need to respect it.

Reply:
Got it. Already put this whole function into loki::worker.